packages/ckeditor5-dev-release-tools/lib/tasks/reassignnpmtags.js

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const { tools } = require( '@ckeditor/ckeditor5-dev-utils' );
+
+/**
+ * Used to switch the tags from `staging` to `latest` for specified array of packages.
+ *
+ * @param {String} authorizedUser User that is authorized to release ckeditor5 packages.
+ * @param {String} version Specifies the version of packages to reassign the tags for.
+ * @param {Array.<String>} packages Array of packages' names to reassign tags for.
+ * @returns {Promise}
+ */
+module.exports = async function reassignNpmTags( { authorizedUser, version, packages } ) {
+	const errors = [];
+	const packagesSkipped = [];
+	const packagesUpdated = [];
+
+	await verifyLoggedInUserIsAuthorizedToPublish( authorizedUser );
+
+	for ( const packageName of packages ) {
+		try {
+			const latestVersion = await exec( `npm show ${ packageName }@latest version` );
+
+			if ( latestVersion === version ) {
+				packagesSkipped.push( `${ packageName }@${ version }` );
+
+				continue;
+			}
+
+			await exec( `npm dist-tag add ${ packageName }@${ version } latest` );
+			packagesUpdated.push( `${ packageName }@${ version }` );
+			await exec( `npm dist-tag rm ${ packageName }@${ version } staging` );
```

Comment:
If in the first script execution the `npm dist-tag rm ckeditor5-foo@1.0.0 staging` throws an error for the package `ckeditor5-foo`, then the next script execution will not call the `npm dist-tag rm ckeditor5-foo@1.0.0 staging` command again.

I don't know if that's a problem, and I don't know if `npm dist-tag rm ...` is even needed. Probably `ckeditor5-foo@latest` and `ckeditor5-foo@staging` will point to the same version in that case.

Reply:
We agreed to remove `npm dist-tag rm` because it makes no difference for the published package, and we would have to add logic with error handling and retriability for it.